utilities/apache-spark-agents/spark-troubleshooting-agent-notification-integration/SparkTroubleshootingAgentNotificationIntegration_README.md

```diff
@@ -0,0 +1,354 @@
+
+Click the "Confirm subscription".
+
+### Step 2. Build the Lambda Deployment Package (Optional)
+
+In case you’d like to make additional custom changes to the provided [Lambda function](https://code.amazon.com/packages/WetangPythonWebApps/blobs/2c40323f0945bdc9dd362a4ffbdb24c9b163864d/--/lambda_function.py#L177,L178,L230,L231), you can build it by running the [build_lambda_package.sh](https://code.amazon.com/packages/WetangPythonWebApps/blobs/heads/eventbridge_lambda_email/--/build_lambda_package.sh) script. This script creates a ZIP file with all dependencies needed for the Lambda deployment; Otherwise, you can use the provided deployment package ZIP File in the GitHub repository directly.
+
+
+### Step 3. Upload the Lambda Deployment Package to your S3 bucket
+
+```
+*# Replace YOUR_BUCKET with your S3 bucket name*
+aws s3 cp spark-analysis-lambda.zip \
+    s3://YOUR_BUCKET/lambda-packages/spark-analysis-lambda.zip
+```
+
+### Step 4. Create the Lambda function execution role
+
+The Lambda function requires an IAM role with specific permissions. Create the role with the following policies:
+
+#### 4.1: Create Trust Policy
+
+Create a file named `lambda-trust-policy.json`:
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "lambda.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+
+```
+
+#### 4.2: Create the IAM Role
+
+```
+aws iam create-role \
+    --role-name spark-analysis-lambda-role \
+    --assume-role-policy-document file://lambda-trust-policy.json
+```
+
+#### 4.3: Attach AWS Managed Policy for Basic Lambda Execution
+
+This provides CloudWatch Logs permissions:
+
+```
+aws iam attach-role-policy \
+    --role-name spark-analysis-lambda-role \
+    --policy-arn arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+```
+
+#### 4.4: Create and Attach SMUS MCP Policy
+
+Create a file named `smus-mcp-policy.json`:
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+        "Sid": "UsingSMUSMCP",
+        "Effect": "Allow",
+        "Action": [
+            "sagemaker-unified-studio-mcp:InvokeMcp",
+            "sagemaker-unified-studio-mcp:CallReadOnlyTool",
+            "sagemaker-unified-studio-mcp:CallPrivilegedTool"
+        ],
+        "Resource": "*"
+        }
+    ]
+}
+```
+
+Attach the policy:
+
+```
+aws iam put-role-policy \
+    --role-name spark-analysis-lambda-role \
+    --policy-name SMUSMCPPolicy \
+    --policy-document file://smus-mcp-policy.json
+```
+
+#### 4.5: Add SNS Publish Permission
+
+Create a file named `sns-publish-policy.json`: (replace REGION and YOUR_ACCOUNT) 
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "sns:Publish"
+            ],
+            "Resource": "arn:aws:sns:REGION:YOUR_ACCOUNT:spark-analysis-notifications"
```

Comment:
Permission is namespaced to resource name which is provisioned by CFN, could just replace the REGION/ACCOUNT with `*`